python/ray/data/datasource/parquet_datasource.py

```diff
@@ -314,7 +314,13 @@ def _estimate_files_encoding_ratio(self) -> float:
         futures = []
         for idx, sample in enumerate(file_samples):
             # Sample i-th row group in i-th file.
-            futures.append(sample_piece.remote(_SerializedPiece(sample), idx))
+            # Use SPREAD scheduling strategy to avoid packing many sampling tasks on
+            # same machine to cause OOM issue, as sampling can be memory-intensive.
```

Comment:
Why does this cause an OOM? We should read a rowbatch at most which should be pretty small right? Are we using the right read APIs?

Reply:
@ericl - yes that's the expectation. I actually found Arrow reader allocated much more memory than needed in https://github.com/ray-project/ray/issues/26995#issuecomment-1196283417 . The current API `ParquetFileFragment.head(num_rows, batch_size=num_rows)` indeed gives us pretty accurate in-memory data size. So I am not aware of other better APIs, but I will create a Arrow JIRA tomorrow to ask Arrow experts.